File: LeetCode/1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        int counts[2001] =
            {
                0,
            },
            ouccurences[1001] = {
                0,
            };
        for (int num : arr) {
            num += 1000;
            ++counts[num];
        }
        for (int i = 0; i <= 2000; i++) {
            int count = counts[i];
            if (count == 0)
                continue;
            if (ouccurences[count])
                return false;
            ouccurences[count] = 1;
        }
        return true;
    }
};

File: LeetCode/1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        occurences = {}

        for num in arr:
            if num in occurences:
                occurences[num] += 1
            else:
                occurences[num] = 1

        return len(set(occurences.values())) == len(set(occurences.keys()))